roc-for-elm-programmers.md

```diff
@@ -339,28 +339,24 @@ table :
 table = \{ height, width, title ? "", description ? "" } ->
 ```
 
-This says that `table` takes a record with two _required_ fields (`height` and
-`width` and two _optional_ fields (`title` and `description`). It also says that
-the `height` and `width` fields have the type `Pixels` (a type alias for some
-numeric type), whereas the `title` and `description` fields have the type `Str`.
-This means you can choose to omit `title`, `description`, or both, when calling
-the function...but if you provide them, they must have the type `Str`.
+This says that `table` takes a record with two fields that are _required_, `height` and
+`width`, and two fields that may be _omitted_, `title` and `description`. It also says that
+the `height` and `width` fields have the type `Pixels`, a type alias for some
+numeric type, and the `title` and `description` fields have the type `Str`.
+This means you can choose to omit the `title`, `description`, or both fields, when calling
+the function... but if you provide them, they must have the type `Str`.
 
 This is also the type that would have been inferred for `table` if no annotation
 had been written. Roc's compiler can tell from the destructuring syntax
-`title ? ""` that `title` is an optional field, and that it has the type `Str`.
-These default values can reference other expressions in the record destructure; if you
-wanted, you could write
-`{ height, width, title ? "", description ? Str.concat "A table called " title }`.
-
-Destructuring is the only way to implement a record with optional fields.
-(For example, if you write the expression `config.title` and `title` is an
-optional field, you'll get a compile error.)
-
-This means it's never possible to end up with an "optional value" that exists
-outside a record field. Optionality is a concept that exists only in record
-fields, and it's intended for the use case of config records like this. The
-ergonomics of destructuring mean this wouldn't be a good fit for data modeling.
+`title ? ""` that `title` is a default value field, and that it has the type `Str`.
+
+Destructuring is the only way to implement a record with default value fields. For example,
+if you write the expression `config.title` and `title` is a default value field,
+you'll get a compile error.
+
+Default values is a concept that exists only in record fields, and it's intended for the use
+case of config records like this. The ergonomics of destructuring mean this wouldn't be a good
+fit for data modeling, consider using a `Result` type instead.
```

Comment:
Note, this section has larger changes cause I synced it with the tutorial wording. Looks like it missed some updates that made it into the tutorial section.